### PR TITLE
provider/docker: Paginage catalog results

### PR DESCRIPTION
**Pull request by lwander**
@tomaslin

### CODE REVIEW

**Comment by lwander** [comment]
Tested on a local docker registry with a paginate size of `n=1,2,3` for a handful of repositories.

File: clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy [hunk 1]
```diff
@@ -49,6 +51,9 @@ class DockerRegistryClient {
   @Value('${dockerRegistry.client.timeout:60000}')
   int clientTimeout
 
+  @Value('${dockerRegistry.client.paginateSize:100}')
+  int paginateSize
```
Comment (lwander) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: @tomaslin, I actually can't seem to get this value to be read properly. Regardless of if I have a `dockerRegistry.client.paginateSize` field in my config, it's always set to 0. I had to manually override it for testing. Any ideas?

Comment (tomaslin) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: what are you putting in your config?

Comment (lwander) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]:
```
dockerRegistry:
  client:
    paginateSize: 1
```

Even without that, it should default to 100, right?

Comment (gregturn) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: If it was loaded through Spring, yes. If not, Java would default an int to 0.

Comment (tomaslin) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: yeah, the dockerRegistryClient is not being loaded with Spring it seems

Comment (gregturn) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/client/DockerRegistryClient.groovy]: DockerRegistryClient has no @Component-based annotations, so unless it is being explicitly configured somewhere via @bean, it won't get picked up by component scanning.

**Comment by tomaslin** [comment]
@lwander any updates on this? feels the solution would be to move the values you're injecting into the providerconfig or make the client into a proper spring bean.

**Comment by lwander** [comment]
Whoops, forgot about this. I'll take a look now.

File: clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/config/DockerRegistryConfigurationProperties.groovy [hunk 2]
```diff
@@ -37,6 +37,10 @@ class DockerRegistryConfigurationProperties {
     String address
     // How many threads to cache all provided repos on. Really only useful if you have a ton of repos.
     int cacheThreads
+    // Timeout time in milliseconds for this repository. Default is 60,000 (1 minute).
+    long clientTimeoutMillis
+    // Paginate size for the docker repository /_catalog endpoint. Default is 100
+    int paginateSize
```
Comment (lwander) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/config/DockerRegistryConfigurationProperties.groovy]: @tomaslin I moved clientTimeout into clientTimeoutMillis to per-account configuration. Getting the beans to work was too much of a pain, as the autowired dependencies aren't available until after the constructor runs.

Comment (tomaslin) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/config/DockerRegistryConfigurationProperties.groovy]: SGTM

Comment (lwander) [inline clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/config/DockerRegistryConfigurationProperties.groovy]: Cool, going to rebase and merge this then.

**Pull request merged**